test/test_vmap.py

```diff
@@ -2719,14 +2719,31 @@ def naive_f(x, shape):
 
         self.assertTrue(torch.randn(()).dim() == 0)
 
-    @parametrize('op', [torch.cos, torch.sinh], name_fn=lambda f: f.__name__)
-    def test_foobar_parametrize(self, op):
-        pass
+    @parametrize('in_dim', [0, 1, 2])
+    @parametrize('out_dim', [0, 1, 2])
+    @parametrize('randomness', ['error', 'different', 'same'])
+    def test_chunk_vmap(self, in_dim, out_dim, randomness):
+
+        if (in_dim > 0 or out_dim > 0) and randomness == "different":
```

Comment:
I have an impression that vmap with randomness="different" and in_dim > 0 is not working as expected:
```python
in_dim = 1
out_dim = 0
randomness = "different"

torch.manual_seed(1)
x = torch.randn(4, 5)

def f(x):
    y = x.sin() + torch.rand_like(x)
    return y

rs = torch.get_rng_state()

expected = ft.vmap(f, in_dims=in_dim, out_dims=out_dim, randomness=randomness)(x)
print("expected:", expected)

xs = x.split(1, dim=in_dim)
torch.set_rng_state(rs)

out = []
for x in xs:
    out.append(ft.vmap(f, in_dims=in_dim, out_dims=out_dim, randomness=randomness)(x))
result = torch.cat(out, out_dim)
print("output:", result)
```
Output:
```
expected: tensor([[-0.0599,  0.5304,  1.2153,  0.1304],
        [-0.2651,  0.2595,  0.5518,  0.0550],
        [ 0.1056,  1.2283,  0.4167,  0.7079],
        [-0.7316,  0.6618, -0.1340,  1.2699],
        [ 1.7522,  1.6782, -0.6967,  1.8902]])
output: tensor([[-0.0599,  0.6587,  1.6947,  0.1364],
        [ 0.3088, -0.3270,  0.7797, -0.2103],
        [-0.3718,  1.4794,  0.2957,  0.9765],
        [-0.6436,  0.8703, -0.5599,  0.5335],
        [ 1.5329,  1.2997,  0.2820,  1.8902]])
```

However, results are the same if `in_dim=0`. Maybe, i should miss something...

cc @samdow

Reply:
This is expected I think. The batch rule for randn_like moves the batch dim to the front before calling randn_like: https://github.com/pytorch/functorch/blob/fb70a3cb588c4b937bb21479de762179b61b7661/functorch/csrc/BatchRulesRandomness.cpp#L164 . So the tensor that gets generated by using a shape of (N, 5) is different from the numbers that get generated by using a shape of (5, N) (the numbers two tensors will be transposed of each other).

In general, vmap(f) with randomness="different" does not promise the same semantics as doing a for-loop over f because the operations get executed in a different sequence, thereby advancing the global RNG state differently